taichi/ir/control_flow_graph.cpp

```diff
@@ -501,7 +501,8 @@ bool CFGNode::dead_store_elimination(bool after_lower_access) {
       }
     }
     auto load_ptrs = irpass::analysis::get_load_pointers(stmt);
-    if (load_ptrs.size() == 1 && store_ptrs.empty() && stmt->width() == 1) {
+    if (load_ptrs.size() == 1 && store_ptrs.empty() && stmt->width() == 1 &&
+        !stmt->is<PtrOffsetStmt>()) {
```

Comment:
Was wondering how "PtrOffsetStmt" get related to MatrixType change?

Reply:
I would also like to understand this change. Why do we exclude `PtrOffsetStmt`? If this only cannot work under `real_matrix=True`, we should check `real_matrix` as well.

Reply:
I was trying to adapt CFG pass with matrix indexing. With the current implementation, this could be removed. (will be in a separate PR)